src/tracks/playabletrack/wavetrack/ui/WaveformView.cpp

```diff
@@ -456,7 +456,7 @@ void DrawIndividualSamples(TrackPanelDrawingContext &context, size_t channel,
    const auto &zoomInfo = *artist->pZoomInfo;
 
    const double toffset = clip->GetPlayStartTime();
-   double rate = clip->GetRate();
+   const double rate = clip->GetRate() / clip->GetStretchRatio();
```

Comment:
I see this change in `DrawIndividualSamples`.  I wonder whether it's complete.  I see another use of `clip->GetRate()` that is not corrected for stretch later.  But it's only fetching one value from the envelope and passing a time-step value for successive samples that doesn't matter and could be 0.

Reply:
The domain of the Envelope (viewed as a curve or function) is always adjusted to be real time, correct?  So `time` given to `GetValue` is right?

Did you test display of individual samples with an envelope too?

Reply:
> The domain of the Envelope (viewed as a curve or function) is always updated with ration changes to be real time, correct?

Correct. The x-position of the samples is scaled by the stretch ratio because the audio samples are stored in stretched state. The envelope is, though.

> Did you test display of individual samples with an envelope too?

I did, yes, and that works to the extent I tested it. Not so confident about the corner cases, though.